lapce-data/src/buffer.rs

```diff
@@ -1459,6 +1459,12 @@ impl Buffer {
                     self.offset_of_line_col(line, col, config.editor.tab_width);
                 (new_offset, horiz)
             }
+            Movement::DocumentStart => {
+                (0, ColPosition::Start)
+            }
+            Movement::DocumentEnd => {
+                (self.len(), ColPosition::End)
```

Comment:
For normal mode, you need the offset to be self.len() -1

Reply:
Actually it's a bit more complicated than that. For Vim, if the document ends with \n, then it will be self.len(), otherwises, move to previous grapheme boundary of self.len()